### PR TITLE
AISDK-107: Add examples to the java SDK

### PR DESCRIPTION
**Pull request by Chanc3**
This PR also:
* Fixes the flakiness of the custom vocab delete test
* Added the filter profanity option to async requests

### CODE REVIEW

**Comment by Chanc3** [comment]
A few questions I have are:
- Do we need to include audio files to use in the examples? (Python does not but Node does)
- Should I include examples for GET /jobs and GET /account? (These two don't really seem necessary)

File: examples/AsyncTranscribeMediaUrl.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    revAiJobOptions.setMetadata("My first submission");
+    revAiJobOptions.setCallbackUrl("https://example.com");
+    revAiJobOptions.setSkipPunctuation(false);
+    revAiJobOptions.setSkipDiarization(false);
+    revAiJobOptions.setFilterProfanity(true);
+    revAiJobOptions.setSpeakerChannelsCount(null);
+
+    RevAiJob submittedJob;
+
+    String mediaUrl = "https://your.media.url";
```
Comment (amikofalvy) [inline examples/AsyncTranscribeMediaUrl.java]: You can use the example media url we use in other places for rev.ai. It contains a real media file.
`https://support.rev.com/hc/en-us/article_attachments/200043975/FTC_Sample_1_-_Single.mp3`

File: examples/AsyncTranscribeMediaUrl.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      throw new RuntimeException("Failed to submit url [" + mediaUrl + "] " + e.getMessage());
+    }
+    String jobId = submittedJob.getJobId();
+    System.out.println("Job Id: " + jobId);
+    System.out.println("Job Status: " + submittedJob.getJobStatus());
+    System.out.println("Created On: " + submittedJob.getCreatedOn());
+
+    /*
+     * Waits 5 seconds between each status check to see if job is complete.
+     * NOTE: polling for job status is not recommended in a non-testing environment.
```
Comment (amikofalvy) [inline examples/AsyncTranscribeMediaUrl.java]: We don't disallow or recommend against polling for all customers. I would remove this line and description of best practices for when to use webhooks.

File: examples/CustomVocabulariesExample.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    String customVocabularyId = submittedCustomVocabularyInfo.getId();
+    System.out.println("Vocabulary Id: " + customVocabularyId);
+    System.out.println("Vocabulary Status: " + submittedCustomVocabularyInfo.getStatus());
+    System.out.println("Created On: " + submittedCustomVocabularyInfo.getCreatedOn());
+
+    /*
+     * Waits 5 seconds between each status check to see if vocabulary has been processed.
+     * NOTE: polling for status is not recommended in a non-testing environment.
```
Comment (amikofalvy) [inline examples/CustomVocabulariesExample.java]: ditto

File: src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java [hunk 4]
```diff
@@ -44,6 +44,10 @@
   @SerializedName("metadata")
   private String metadata;
 
+  /** Optional parameter to filter profanity in the transcript */
+  @SerializedName("filter_profanity")
```
Comment (amikofalvy) [inline src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java:48]: I think it makes sense to add these in a separate PR.
Landing logically separate improvements as separate commits allows us to easily rollback changes with git cli and provides a better git history with clear commit messages.

Comment (amikofalvy) [reply src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java:48]: for example: we could also add remove_disfluencies option, but I think it would be best if that happened in a separate commit

Comment (amikofalvy) [reply src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java:48]: But if filter_profanity is already implemented in this pr then no reason to delay by taking it out. we'll just keep it in mind for the future.

**Review by amikofalvy — APPROVED** [review]
> Do we need to include audio files to use in the examples? (Python does not but Node does)

I think it is fine without files for local submission. I did suggest that you change the media_urls to a valid url.

> Should I include examples for GET /jobs and GET /account? (These two don't really seem necessary)

I don't think these two endpoints require examples.

Once all feedback is addressed, and I'm giving the PR a :shipit: 

Please have a third party verify the examples (Raabya). It is critical that these examples work and are useable by someone other than the author. QA can help us verify the examples work as they expect.
